addons/mail/models/discuss/discuss_channel.py

```diff
@@ -336,8 +371,8 @@ def create(self, vals_list):
             # is_pinned + ensure they have rights to see channel
             if not self.env.context.get('install_mode') and not self.env.user._is_public():
                 partner_ids_to_add = list(set(partner_ids + [self.env.user.partner_id.id]))
-            vals['channel_member_ids'] = membership_ids_cmd + [
-                (0, 0, {'partner_id': pid})
+            vals["channel_member_ids"] = membership_ids_cmd + [
+                (0, 0, {"partner_id": pid, "channel_role": "owner" if vals.get("channel_type") in ["channel", "group"] and pid == self.env.user.partner_id.id else None})
```

Comment:
The more I think about it, I believe the owner feature should be exclusively for "group" type.

It feels like "channel" should be managed by DB admin (or if we introduce it, discuss admin) instead.

I'm not sure though, I guess both could co-exist. But it feels for channel we should then have a distinction between "user-owned" channels and "db-owned" channels.

But I guess DB admin should be able to just remove owner from them if they don't want a channel to be "owned".